packages/devtools-network-console/src/ui/JSONView/index.tsx

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import React from 'react';
+import { TreeView, TreeViewItem } from '@microsoft/fast-components-react-msft';
+import { LocalizationConsumer, getText } from 'utility/loc-context';
+import Property from './Property';
+
+interface IJSONViewProps {
+    value: any;
+}
+
+function produceChildren(value: any, depth: number) {
+    if (typeof value !== 'object' || value === null) {
+        return undefined;
+    }
+
+    let children: Array<React.ReactNode>;
+    const isArray = Array.isArray(value);
+    if (isArray) {
+        children = Array.from(produceArrayChildren(value, depth));
+    }
+    else {
+        children = Array.from(produceObjectChildren(value, depth));
+    }
+    return children;
+}
+
+function* produceObjectChildren(value: object, depth: number): IterableIterator<React.ReactNode> {
+    const keys = Object.keys(value);
+    for (let i = 0; i < keys.length; i++) {
+        const key = keys[i];
+        const childValue = (value as Record<string, unknown>)[key];
+        if (typeof childValue === 'object' && childValue !== null && Object.keys(childValue).length > 0) {
+            yield <TreeViewItem
+                        key={key}
+                        className="json-tree-view-item-can-wrap"
+                        titleContent={<Property keyKind="string" name={key} value={childValue} />}
+                        children={produceChildren(childValue, depth + 1)}
+                        aria-level={depth}
+                        />;
+        }
+        else {
+            yield <TreeViewItem
+                        key={key}
+                        className="json-tree-view-item-can-wrap"
+                        titleContent={<Property keyKind="string" name={key} value={childValue} />}
+                        aria-level={depth}
+                        />;
```

Comment:
Since these two JSX objects here are identical except for the `children` prop, suggest you use the JSX spread operator here so there is less duplication. E.g.

```
const itemProps: TreeViewItemProps = {
  key,
  className: "json-tree-view-item-can-wrap",
  titleContent: (<Property keyKind="string" name={key} value={childValue} />)
  aria-level: depth
}

...

yield <TreeViewItem {...itemProps} children={produceChildren(childValue, depth + 1)}>
// alternatively, logically assign children when declaring props, then just use spread operator
```